e2e/testcases/override_resource_limits_test.go

```diff
@@ -151,33 +103,44 @@ func TestOverrideReconcilerResourcesV1Alpha1(t *testing.T) {
 	nt.MustMergePatch(rootSync, `{"spec": {"override": {"resources": [{"containerName": "reconciler", "cpuRequest": "500m", "cpuLimit": "800m", "memoryRequest": "400Mi", "memoryLimit": "411Mi"}]}}}`)
 	rootReconcilerDeploymentGeneration++
 
+	updatedRootReconcilerResources := v1beta1.ContainerResourcesSpec{
+		ContainerName: reconcilermanager.Reconciler,
+		CPURequest:    resource.MustParse("500m"),
+		CPULimit:      resource.MustParse("800m"),
+		MemoryRequest: resource.MustParse("400Mi"),
+		MemoryLimit:   resource.MustParse("411Mi"),
+	}
+
 	// Verify the reconciler container of root-reconciler uses the new resource request and limits, and the git-sync container uses the default resource requests and limits.
 	err = nt.Watcher.WatchObject(kinds.Deployment(),
 		rootReconcilerNN.Name, rootReconcilerNN.Namespace,
 		[]testpredicates.Predicate{
 			testpredicates.GenerationEquals(rootReconcilerDeploymentGeneration),
-			testpredicates.HasCorrectResourceRequestsLimits(reconcilermanager.Reconciler, resource.MustParse("500m"), resource.MustParse("800m"), resource.MustParse("400Mi"), resource.MustParse("411Mi")),
-			testpredicates.HasCorrectResourceRequestsLimits(reconcilermanager.GitSync, defaultGitSyncCPURequest, defaultGitSyncCPULimits, defaultGitSyncMemRequest, defaultGitSyncMemLimits),
+			testpredicates.DeploymentContainerResourcesEqual(updatedRootReconcilerResources),
+			testpredicates.DeploymentContainerResourcesEqual(defaultResources[reconcilermanager.GitSync]),
 		},
-		testwatcher.WatchTimeout(30*time.Second))
+		testwatcher.WatchTimeout(30*time.Second),
```

Comment:
I suspect 30 seconds may not always be long enough for autopilot

Reply:
This is just the Deployment spec update, not the Pod update. The reconciler Deployment should get updated immediately by the reconciler-manager when the RootSync spec update event is received. So I don't know if being on autopilot matters, unless the reconciler-manager itself is being rescheduled.

Reply:
Yeah my main concern is in the case of reconciler-manager being rescheduled. Perhaps unlikely but could be a source of flakiness

Reply:
removed